pkg/scm/git/git_test.go

```diff
@@ -230,9 +230,22 @@ func TestMungeNoProtocolURL(t *testing.T) {
 			}
 		}
 		if !equal {
-			t.Errorf("For URL string %s, field by field check for scheme %v opaque %v host %v path %v rawq %v frag %v out user nil %v test user nil %v out scheme  %s out opaque %s out host %s out path %s  out raw query %s out frag %s", scenario,
-				uri.Scheme == test.Scheme, uri.Opaque == test.Opaque, uri.Host == test.Host, uri.Path == test.Path, uri.RawQuery == test.RawQuery,
-				uri.Fragment == test.Fragment, uri.User == nil, test.User == nil, uri.Scheme, uri.Opaque, uri.Host, uri.Path, uri.RawQuery, uri.Fragment)
+			t.Errorf(`URL string %q, field by field check:
+- Scheme: got %v, ok? %v
+- Opaque: got %v, ok? %v
+- Host: got %v, ok? %v
+- Path: got %v, ok? %v
+- RawQuery: got %v, ok? %v
+- Fragment: got %v, ok? %v
+- User: got %v`,
+				scenario,
+				uri.Scheme, uri.Scheme == test.Scheme,
+				uri.Opaque, uri.Opaque == test.Opaque,
+				uri.Host, uri.Host == test.Host,
+				uri.Path, uri.Path == test.Path,
+				uri.RawQuery, uri.RawQuery == test.RawQuery,
+				uri.Fragment, uri.Fragment == test.Fragment,
+				uri.User)
```

Comment:
I don't think anybody was reading the previous string...

The way to go is to replace that with the simpler `reflect.DeepEqual` code once we drop support for Go 1.4. Been there in origin and reverted because of language differences: https://github.com/openshift/origin/pull/6402, https://github.com/openshift/origin/pull/6578.